drivers/i2c/i2c_ite_it8xxx2.c

```diff
@@ -309,6 +323,373 @@ static void i2c_w2r_change_direction(const struct device *dev)
 	}
 }
 
+#ifdef CONFIG_I2C_IT8XXX2_FIFO_MODE
+static void i2c_fifo_en_w2r(bool enable, int port)
+{
+	unsigned int key = irq_lock();
+
+	if (enable) {
+		if (port == 0) {
+			IT8XXX2_SMB_I2CW2RF |= IT8XXX2_SMB_MAIF |
+					       IT8XXX2_SMB_MAIFI;
+		} else if (port == 2) {
+			IT8XXX2_SMB_I2CW2RF |= IT8XXX2_SMB_MBCIF |
+					       IT8XXX2_SMB_MCIFI;
+		}
+	} else {
+		if (port == 0) {
+			IT8XXX2_SMB_I2CW2RF &= ~(IT8XXX2_SMB_MAIF |
+						 IT8XXX2_SMB_MAIFI);
+		} else if (port == 2) {
+			IT8XXX2_SMB_I2CW2RF &= ~(IT8XXX2_SMB_MBCIF |
+						 IT8XXX2_SMB_MCIFI);
+		}
+	}
+
+	irq_unlock(key);
+}
+
+static int i2c_tran_fifo_write_to_read(const struct device *dev)
+{
+	struct i2c_it8xxx2_data *data = dev->data;
+	const struct i2c_it8xxx2_config *config = dev->config;
+	uint32_t i;
+	uint8_t *base = config->base;
+	volatile uint8_t *reg_mstfctrl = config->reg_mstfctrl;
+
+	if (data->msgs->flags & I2C_MSG_START) {
+		/* Clear start flag. */
```

Comment:
There appears to be a duplicate of the `I2C_MSG_START`  block from `i2c_tran_fifo_write()`.  If you create the following helper functions, I think you can save some code space, and improve readabily:

```
i2c_tran_fifo_write_start()  /* I2C_MSG_START path */
i2c_tran_fifo_write_next_block() /* else if SMB_BLKDS path */
i2c_tran_fifo_write_finish()
i2c_tran_fifo_read_start()
i2c_tran_fifo_read_next_block()
i2c_tran_fifo_read_finish()
```

This function then becomes:
```
if (data->msgs->flags & START)
    i2c_tran_fifo_write_start()
else
    if (IT8XXX2_SMB_BLKDS)
        if (IT8XXX2_SMB_I2C_SW_EN)
            i2c_tran_fifo_read_next_block()
        else
            i2c_tran_fifo_write_next_block()
    else if (IT8XXX2_SMB_IWRFISTA)
        i2c_tran_fifo_read_start()
    else
        i2c_tran_fifo_read_finish()
```

Reply:
Thanks for your suggestion.